ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVTest.scala

```diff
@@ -276,4 +257,143 @@ object KVTest {
       }
     } yield result
 
+  val minMRTDelta: Duration = Duration.ofSeconds(1)
+
+  private[kvutils] val metrics = new Metrics(new MetricRegistry)
+
+  private def submit(submission: DamlSubmission): KVTest[(DamlLogEntryId, DamlLogEntry)] =
+    for {
+      testState <- get[KVTestState]
+      entryId <- freshEntryId
+      (logEntry, newState) = keyValueCommitting.processSubmission(
+        entryId = entryId,
+        recordTime = testState.recordTime,
+        defaultConfig = testState.defaultConfig,
+        submission = submission,
+        participantId = testState.participantId,
+        inputState = submission.getInputDamlStateList.asScala.map { key =>
+          key -> testState.damlState.get(key)
+        }.toMap,
+      )
+      _ <- addDamlState(newState)
+    } yield {
+      // Verify that all state touched matches with "submissionOutputs".
+      assert(
+        newState.keySet subsetOf keyValueCommitting.submissionOutputs(submission)
+      )
+      // Verify that we can always process the log entry
+      val _ = KeyValueConsumption.logEntryToUpdate(entryId, logEntry)
+
+      entryId -> logEntry
+    }
+
+  private def preExecute(
+      damlSubmission: DamlSubmission): KVTest[(DamlLogEntryId, PreExecutionResult)] =
+    for {
+      testState <- get[KVTestState]
+      entryId <- freshEntryId
+      preExecutionResult @ PreExecutionResult(_, successfulLogEntry, newState, _, _, _, _) = keyValueCommitting
+        .preExecuteSubmission(
+          defaultConfig = testState.defaultConfig,
+          submission = damlSubmission,
+          participantId = testState.participantId,
+          inputState = damlSubmission.getInputDamlStateList.asScala.map { key =>
+            {
+              val damlStateValue = testState.damlState
+                .get(key)
+              key -> (damlStateValue -> damlStateValue
+                .map(_.toByteString)
+                .getOrElse(FingerprintPlaceholder))
+            }
+          }.toMap,
+        )
+      _ <- addDamlState(newState)
+    } yield {
+      // Verify that we can always process the log entry
+      val _ =
+        KeyValueConsumption.logEntryToUpdate(
+          entryId,
+          successfulLogEntry,
+          recordTimeFromTimeUpdateLogEntry)
+
```

Comment:
Please verify that the rejection log entry can also be processed.

Reply:
Done.